python/cudf_polars/tests/test_config.py

```diff
@@ -58,6 +58,16 @@ def test_invalid_memory_resource_raises(mr):
         q.collect(engine=pl.GPUEngine(memory_resource=mr))
 
 
+@pytest.mark.parametrize("disable_uvm", [True, False])
+def test_cudf_polars_enable_disable_uvm(monkeypatch, disable_uvm):
+    q = pl.LazyFrame({"a": [1, 2, 3]})
+
+    with monkeypatch.context() as monkeycontext:
+        monkeycontext.setenv("CUDF_POLARS_DISABLE_UVM", str(disable_uvm))
+        result = q.collect(engine=pl.GPUEngine())
+    assert_frame_equal(q.collect(), result)
```

Comment:
I tried looking around for API's/context managers in rmm how to achieve this and found none.

Reply:
Could you use https://docs.rapids.ai/api/rmm/stable/guide/#memory-statistics-and-profiling and then check the string generated in the report?

Reply:
We could at least directly invoke the `default_memory_resource` function and ensure that it returns what we expect. That's a true unit test, though it won't verify that an actual collect call would use that memory resource.

Reply:
> We could at least directly invoke the `default_memory_resource` function and ensure that it returns what we expect. That's a true unit test, though it won't verify that an actual collect call would use that memory resource. It's a relatively simple test though, especially since you can check the upstream_mr property for nested resources.

Used this approach to test 👍